### PR TITLE
VAOS + FHIR - Search Slot Resource (Collection)

### PR DESCRIPTION
**Pull request by ajmagdub**
## Description of change

Add controller, routes, parameter validations, and mock VCR Cassette for specs for the FHIR Service to support Slot resource.

## Original issue(s)
[#9097 VAOS + FHIR - Search Slot Resource](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/9097)

### CODE REVIEW

File: modules/vaos/spec/request/v1/slot_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        expect(error_object['resourceType']).to eq('Slot')
+        expect(error_object['issue'].first['details']['text'])
+          .to eq('You do not have access to online scheduling')
+      end
+    end
+
+    context 'with a loa3 user' do
+      let(:user) { FactoryBot.create(:user, :vaos) }
+
+      it 'and no slots found' do
```
Comment (kreek) [inline modules/vaos/spec/request/v1/slot_request_spec.rb:29]: Just semantics but I think the 'and' here makes the `it` seem more like a `context`, maybe wrap it in another context block:
``` ruby
context 'with an empty response' do
  it 'returns no slots' do
```

File: modules/vaos/spec/request/v1/slot_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        VCR.use_cassette('vaos/fhir/slot/search_200_no_slots_found', match_requests_on: %i[method uri]) do
+          get '/vaos/v1/Slot?schedule=no-such-resource'
+          expect(response).to have_http_status(:ok)
+          expect(JSON.parse(response.body)['total']).to eq(0)
+        end
+      end
+
+      it 'and slots found' do
```
Comment (kreek) [inline modules/vaos/spec/request/v1/slot_request_spec.rb:37]: same as above

File: modules/vaos/spec/request/v1/slot_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        VCR.use_cassette('vaos/fhir/slot/search_200_slots_found', match_requests_on: %i[method uri]) do
+          get '/vaos/v1/Slot?schedule=Schedule/sch.nep.AVT987.20191208'
+          expect(response).to have_http_status(:ok)
+          expect(JSON.parse(response.body)['total']).to eq(2)
+          expect(JSON.parse(response.body)['entry'][0]['resource']['start']).to eq('2019-12-08T09:15:00Z')
+          expect(JSON.parse(response.body)['entry'][1]['resource']['start']).to eq('2019-12-08T09:15:00Z')
+        end
+      end
+
+      it 'and server returns a 500 error' do
```
Comment (kreek) [inline modules/vaos/spec/request/v1/slot_request_spec.rb:47]: and here 😄

File: modules/vaos/spec/request/v1/slot_request_spec.rb (hunk 2 above)
Comment (ajmagdub) [reply modules/vaos/spec/request/v1/slot_request_spec.rb]: Thanks Alastair, will incorporate your comment

File: modules/vaos/spec/request/v1/slot_request_spec.rb (hunk 3 above)
Comment (ajmagdub) [reply modules/vaos/spec/request/v1/slot_request_spec.rb]: Thanks Alastair, will incorporate your comment

**Pull request merged**